File: agentcore/src/main/java/agentcore/agentcommunication/IAgentCommunication.java
package agentcore.agentcommunication;

import java.io.IOException;

/**
 * Created by user on 21.02.2017.
 */
public interface IAgentCommunication {

    void sendMassege(Message message)  throws IOException;
    void connect(String host, int port) throws IOException;
    void disconnect() throws IOException;
    boolean isConnect();

}

File: agentcore/src/main/java/agentcore/agentcommunication/IAgentCommunication.kt
package agentcore.agentcommunication

import java.io.IOException

/**
 * @author Nikita Gorodilov
 */
interface IAgentCommunication {

    @Throws(IOException::class)
    fun sendMassege(message: Message)

    @Throws(IOException::class)
    fun connect(host: String, port: Int)

    @Throws(IOException::class)
    fun disconnect()

    fun isConnect(): Boolean
}